tools/ci/ci_agent.py

```diff
@@ -159,6 +180,19 @@ def add_arguments(cmd, *args):
     )
     add_argument(cmd_run, "--targets", required=False)
     cmd_run.set_defaults(func="run")
+
+    cmd_code_coverage = subparsers.add_parser("code-coverage")
+    add_arguments(
+        cmd_code_coverage,
+        "--config-file",
+        "--src",
+        "--build-path",
+        "--build-flags",
+        "--c-flags",
+        "--codecov-token",
+    )
```

Comment:
At this point I'm inclined to think we should just take the config as an argument and do the rest through the config or environment variables. It's getting difficult to grapple with all the options.

Reply:
config is one of the argument, we have options to override options in config

Reply:
I'm not convinced that flexibility is required.